### DISCUSSION TITLE
q-table in grid mode row not clickable

### DISCUSSION THREAD

**Discussion by jrphifer03** [quasarframework/quasar#15555] [General - Components / Directives / etc] [answered]
row is clickable when not on mobile. How do I make row clickable on mobile? Thanks in advance! Please see following code snipet:

`<q-table
              title="Students"
              :columns="student_columns"
              :rows="student_rows"
              row-key="id"
              :grid="$q.screen.xs"
            >
              <template #body="props">
                <q-tr :props="props" @click="onRowClick(props.row)">
                  <q-td key="name" :props="props">
                    <strong>{{ props.row.name }} </strong><br />
                    {{ props.row.school }}
                  </q-td>
                  <q-td key="studentid" :props="props">
                    <strong>{{ props.row.student_id }} </strong><br />
                  </q-td>
                  <q-td key="dob" :props="props">
                    <strong>{{ props.row.dob }} </strong><br />
                  </q-td>
                  <q-td key="grade" :props="props">
                    <strong>{{ props.row.grade }} </strong><br />
                  </q-td>
                  <q-td key="gender" :props="props">
                    <strong>{{ props.row.gender }} </strong><br />
                  </q-td>
                  <q-td key="status" :props="props">
                    <strong>{{ props.row.status }} </strong><br />
                  </q-td>
                  <q-td key="contact" :props="props">
                    <strong>{{ props.row.contact }} </strong><br />
                    {{ props.row.relationship }}
                  </q-td>
                  <q-td key="phone" :props="props">
                    {{ props.row.home_phone }} <br />
                    {{ props.row.cell_phone }} <br />
                    {{ props.row.work_phone }} <br />
                  </q-td>
                </q-tr>
              </template>
            </q-table>`

**Comment by dvbeek** [comment, marked as answer]
Have you tried using the quasar event `row-click`?

I don't think there's a need to define all the cells manually, you can set and style them using the column definitions.
So than you can just do something like:
```
<q-table title="Students" :columns="student_columns" :rows="student_rows" row-key="id" :grid="$q.screen.xs"
  @row-click="filterTable(evt, row, index)" />
```
For details, check https://next.quasar.dev/vue-components/table#qtable-api

**Comment by jrphifer03** [comment]
I need to define the cells manually because I am going to add check boxes
to the row and some of the columns are multiple rows, and some are not.

**Reply by dvbeek** [reply]
You can define the cell manually for the check boxes, there's no need to override the row for that.

I.e.: `<template v-slot:body-cell-action="props">`

To combine multiple values you can do the same, but you can also use `format`:
```
const column = {
    name: 'phone',
    label: 'Phone',
    field: 'home_phone',
    format: formatPhone
  }

function formatPhone(value, row): string {
  return `${row.home_phone}<br>${row.cell_phone}<br>${row.work_phone}<br>
}
```

**Comment by MilosPaunovic** [comment]
Provide a Codepen reproduction of the issue.

**Comment by jrphifer03** [comment]
I will try that. Thanks for your help.

**Comment by jrphifer03** [comment]
That worked.  Thanks for your help!
